Moved Syntax Highlighting page from openhab1-addons wiki

Confectrician opened the pull request:
This is a quick n dirty copy of the according wiki page from the openhab1-addons repo.

I have added basic stuff like the jekyll layout und include tags 
and have also switched the table of contents from Markdown urls to jekyll-toc-tag.
Finally i added an early version of the vscode section for this article.
(Your Feedback is welcome for the vscode section @kubawolanin 👍 )

Feel free to give feedback and improvement suggestions.
As i said its just the quick n dirty way to make some progress for this page.

Open Question for me:
1. What should we do with the currently broken screenshots?
Remove? Make new ones?
2. Do i need to mention or sign off (Also by: XYZ) the existing wiki authors anywhere? 
I am not sure about this.

Fixes #408

Signed-off-by:  Jerome Luckenbach <github@luckenba.ch> (github: @Confectrician)

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+
+![Items](http://wiki.openhab-samples.googlecode.com/hg/screenshots/syntaxhl_npp_items.png "Items")
+![Rules](http://wiki.openhab-samples.googlecode.com/hg/screenshots/syntaxhl_npp_rules.png "Rules")
+![Sitemap](http://wiki.openhab-samples.googlecode.com/hg/screenshots/syntaxhl_npp_sitemap.png "Sitemap")
```
Dead URLs.
We might want to move them to this repository from this thread: https://groups.google.com/forum/#!topic/openhab/VXdBiaY8XB8

Confectrician replied:
Thats one of my above mentioned questions. 
Seems that the currently linked screenshots are lost.

We could remove them for now and see if someone makes new ones in the future.

kubawolanin commented on configuration/editors.md:
````diff
@@ -0,0 +1,164 @@
+```
+
+#### Manual installation
+- Download the syntax files from https://github.com/cyberkov/openhab-vim
````
`[openhab-vim](https://github.com/cyberkov/openhab-vim)`

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+
+#### Manual installation
+- Download the syntax files from https://github.com/cyberkov/openhab-vim
+- place them in your home directory under ~/.vim/
```
please wrap ``~/.vim/`` with single tick `

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+Nano is a common editor in linux systems
+
+### Installing the syntax-files
+- Download the syntax file openhab.nanorc from https://github.com/airix1/openhabnano
```
`[openhabnano](https://github.com/airix1/openhabnano) repository`

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+
+### Installing the syntax-files
+- Download the syntax file openhab.nanorc from https://github.com/airix1/openhabnano
+- place them in your nanorc directory ie: ~/.nano or /usr/share/nano. Then simply include the openhab.nanorc file in your ~/.nanorc or /etc/nanorc file ie:
```
Please wrap paths with `

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+### Manual installation
+
+- Download the syntax file from https://github.com/GrisoMG/openhab-syntax-textwrangler/blob/master/openhab.plist
+- Copy the file openhab.plist into ~/Library/Application Support/TextWrangler/Language Modules
```
Please wrap it with `

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+
+### Manual installation
+
+- Download the syntax file from https://github.com/GrisoMG/openhab-syntax-textwrangler/blob/master/openhab.plist
```
`[openhab-syntax-textwrangler](https://github.com/GrisoMG/openhab-syntax-textwrangler/blob/master/openhab.plist) repository`

kubawolanin commented on configuration/editors.md:
````diff
@@ -0,0 +1,164 @@
+    ```
+    code --install-extension openhab-0.0.2.vsix
+    ```
+Currently openhab-vscode isn't yet listed in microsofts marketplace, but the contributors are working on it already.
````
`[Microsoft Visual Studio Marketplace](https://marketplace.visualstudio.com/vscode)`

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+
+## Notepad++
+
+Notepad++ Version 6.2 or above is required to support UDL2 (User Defined Language v2).
```
I found the GitHub repository copy from the old Google Code: https://github.com/thefrip/openhab-samples

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+- Click "Language" (1)
+- Click "Define your language.." (2)
+- Click "Import..." (3)
+- Select one of the downloaded XML-files
```
Downloaded from here: https://github.com/thefrip/openhab-samples

kubawolanin commented on configuration/editors.md:
```diff
@@ -0,0 +1,164 @@
+### Installing the syntax-files
+
+- download copy the syntax-files to */usr/share/mc/syntax/*
+- https://code.google.com/p/openhab-samples/source/browse/syntaxhl/mc/?repo=wiki
```
Hey, found it! :smiley: 
https://groups.google.com/forum/#!searchin/openhab/syntax$20highlighting%7Csort:relevance/openhab/zJrfTDjahbc/r8GcpKDgr1cJ

Confectrician commented:
Thanks for the feedback @kubawolanin,

I have improved nearly all mentioned parts.
Just for the screenshot thingy i will wait for some more feedback. :)

Confectrician commented:
Hey @ThomDietrich,

I don't want to be impatient, but i have some spare time over the weekend,
so a bit feedback would be nice since i wanted to use this spare time for improvements here.

As already said this PR is just for porting the existing editors page to the docs.
improvements like smart home designer merge can be done afterwards.
(Just my 2 cents)

Bets Regards
Jerome

ThomDietrich commented:
Heyho @Confectrician! Sure thing, don't want to make you wait! I'll merge in the next 24 hours ;)

ThomDietrich commented:
Hey Jerome, I misunderstood your request yesterday. You want feedback, you shall get feedback :)

The first thing catching my eye is, that you didn't use the customized linter rules. The article as it is currently formatted will not render correctly. Jekyll uses a Markdown engine called Kramdown, which has a few extra restrictions. One of those is, that every block element (headlines, code fences, tables,...) needs to be surrounded by empty lines. Another linting issue I can already see is that every code fence has to have an explicit language tag (`sh`, `java`, ...)

In general I think you can go ahead and rework the whole article. Move the vscode project to the top and add the ESH Designer.
Maybe we can move the other editors inside a "Others" section? In general I feel these can be condensed. A short intro-filler-sentence, a link and maybe a screenshot would suffice.

Regarding Notepad++: Instead of giving installation instructions on our documentation page I'd rather see such instructions as a PR of a README.md against the project repo. Wdyt?

I was not sure about the project but since it is still active, we should probably also add https://community.openhab.org/t/beta-tester-for-item-checker-tool-wanted/25481 @kubawolanin wdyt?

Regarding the screenshots: For starters you can just erase them. It would be amazing if you could create new ones for all or a few of these. I can provide some if needed.

Any other concerns I missed? Thanks for your help!!

Confectrician commented:
> The first thing catching my eye is, that you didn't use the customized linter rules. The article as it is currently formatted will not render correctly. Jekyll uses a Markdown engine called Kramdown, which has a few extra restrictions. One of those is, that every block element (headlines, code fences, tables,...) needs to be surrounded by empty lines. Another linting issue I can already see is that every code fence has to have an explicit language tag (sh, java, ...)

Correct, as i said it was was quick n dirty copy PR. :D
I have added some empty lines and added _shell_ as language everywhere now.

> In general I think you can go ahead and rework the whole article. Move the vscode project to the top and add the ESH Designer.
Maybe we can move the other editors inside a "Others" section? In general I feel these can be condensed. A short intro-filler-sentence, a link and maybe a screenshot would suffice.

Ok i will move forward with this. VS Code is on top already.

> Regarding Notepad++: Instead of giving installation instructions on our documentation page I'd rather see such instructions as a PR of a README.md against the project repo. Wdyt?

I think NPP has a good Documentation Homepage if i remember right,
so for me it is fully vald to point to the npp doc article about installing language files.

> Regarding mcedit: The link points to a posting by Christoph Wempe. He is still an active member in the community forum. Could you try to contact him and ask about a repository for these files?

I will try to contact him next days.

> Regarding the screenshots: For starters you can just erase them. It would be amazing if you could create new ones for all or a few of these. I can provide some if needed.

Ok i will remove them for now.

ThomDietrich commented:
Hey @Confectrician, what's the word?

Confectrician commented:
Hey @ThomDietrich,

As said in my last comment:
The kramdown issues should be solved, the vscode section has been moved up
and the other editors have been grouped under an others section.
I didn't have time yet to dig into the esh designer article and move it over.
I am on vacation next week so there will be no progress for that timespan too.

If you are fine with this, we could merge the current state and move esh designer in another PR within august.

Confectrician commented:
Hey @ThomDietrich,

I have updated my branch and added a big commit uncluding the move of the ESH Designer article into the editors page.
Additionally i have contacted Christoph Wempe and will give feedback as soon as i have a response.

I think we are ready for merge, when there are no issues left.
We can add screenshots and small improvements later anyway.
The big work should be done now.

mueller-ma commented on configuration/editors.md:
```diff
@@ -0,0 +1,210 @@
+- Download the UDL2-Files (openHAB-`*`.xml) from [openhab-samples](https://github.com/thefrip/openhab-samples)
+- Install the files like it is described in the [editors documentation](http://docs.notepad-plus-plus.org/index.php/User_Defined_Language_Files#How_to_install_user_defined_language_files)
+
+### vi
```
Is vi capable of syntax files? or just vim and its forks?

Confectrician replied:
Good question and i cant answer it. :D
The initial content is a 100% Fork from https://github.com/openhab/openhab1-addons/wiki/Syntax-Highlighting-for-external-editors

But the download repos readme is mentioning vim only,
so for me it would be ok to change this to "vim"

mueller-ma replied:
I think nowadays its pretty hard to get a pure "vi" with distro repos, because it's mostly a symlink to vim in vi mode

mueller-ma commented on configuration/editors.md:
```diff
@@ -0,0 +1,210 @@
+##### Manual installation
+
+- Download the syntax files from [openhab-vim](https://github.com/cyberkov/openhab-vim)
+- place them in your home directory under `~/.vim/`
```
First letter of a line capital pls

mueller-ma commented on configuration/editors.md:
```diff
@@ -0,0 +1,210 @@
+
+- Download the syntax files from [openhab-vim](https://github.com/cyberkov/openhab-vim)
+- Place them in your home directory under `~/.vim/`
+- Start vim with an openHAB configuration file and it should work.
```
s/vim/Vim/

mueller-ma commented on configuration/editors.md:
```diff
@@ -0,0 +1,210 @@
+
+### nano
+
+Nano is a common editor in linux systems
```
s/linux/Linux

ThomDietrich commented on configuration/editors.md:
````diff
@@ -0,0 +1,188 @@
+    ```
+Currently openhab-vscode isn't yet listed in [Microsoft Visual Studio Marketplace](https://marketplace.visualstudio.com/vscode), but the contributors are working on it already.
+
+[Visit the Extensions GitHub Page for further Informations](https://github.com/openhab/openhab-vscode/blob/master/README.md "GitHub Repo for the VS Code Extension")
````
@kubawolanin how about this alternative way?
https://community.openhab.org/t/openhab-vs-code-extension-0-1-0/31202/5?u=thomdietrich

ThomDietrich commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+
+[Visit the Extensions GitHub Page for further Informations](https://github.com/openhab/openhab-vscode/blob/master/README.md "GitHub Repo for the VS Code Extension")
+
+## Eclipse Smarthome Designer
```
SmartHome

ThomDietrich commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+
+Hint: Press `<Ctrl + N>` to create a new file.
+
+## Others
```
After the longer ESH Designer part, the word "Others" might be unclear. How about "Other Editors" or "Further options" or ...

Confectrician replied:
I have added on explaining sentence anyway because of the cascading headlines without content between thingy.

Wait for the commit ind this should work too now. :)

ThomDietrich commented on configuration/editors.md:
````diff
@@ -0,0 +1,188 @@
+color brightgreen start="/\*" end="\*/"
+color brightgreen start="/\*\*" end="\*/"
+```
+http://notepad-plus-plus.org/news/notepad-6.2-release-udl2.html
````
Missing newline, please turn into a proper markdown link

ThomDietrich replied:
Also, any clue what's the meaning of these lines? The headline just says "comments"...

ThomDietrich commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+
+mcedit is an editor which comes with mc (Midnight Commander).
+
+#### Installing the syntax-files
```
In between two sentences the headline doesn't make much sense and looks horribly empty and alone :)
Could you turn these into three sentences without linebreak?

ThomDietrich commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+
+#### Installing the syntax-files
+
+##### Automatic installation
```
More of a general editors note: Do not cascade multiple headlines without at least a filler sentence. From the looks of it the vim article could live without additional headlines.

ThomDietrich commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+
+## openhab-vscode
+
+openhab-vscode is an extension for the [Visual Studio Code](https://code.visualstudio.com) editor.
```
Could you add a screenshot here? @kubawolanin already contributed a few, I'd prefer a simple png over a gif here for uniformity (I'm ready to be convinced otherwise)

Confectrician replied:
I have looked through the community threads and the repo and only can find gif files.
I will add one for now and we can improve that later. :)

ThomDietrich commented on configuration/editors.md:
````diff
@@ -0,0 +1,188 @@
+
+#### Comments
+
+```shell
````
Go with `ini` for a nicer presentation

```ini
color brightgreen "//.*"
color brightgreen start="/\*" end="\*/"
color brightgreen start="/\*\*" end="\*/"# Comments
color brightgreen "//.*"
color brightgreen start="/\*" end="\*/"
color brightgreen start="/\*\*" end="\*/"# Comments
color brightgreen "//.*"
```

Confectrician commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+
+{% include base.html %}
+
+# Editors - Different ways to power up your textual configuration
```
Wdyt about "simplify" instead of "power up"?

Alternative:
"Different ways to make your textual configuration more comfortable"

(Ofc. Everything with correct headline capitalization then ;) )

ThomDietrich replied:
Not sure, up to you. I'd imagine something simpler like "Editors for your Textual openHAB Configuration"

Confectrician commented on configuration/editors.md:
```diff
@@ -0,0 +1,188 @@
+# Editors - Different ways to power up your textual configuration
+
+There are several existing solutions, that can help you configuring your openHAB instance in a textual way.
+Mainly there are two bigger projects for this. 
```
I will change the sentence above too.
Otherwise we would have 2 sentences beginning with "there" which sounds bad while reading.

Confectrician commented:
Ok i have reworked the whole Page now.

Please have a look at the "Others" Section.
I was a bit mad about the different amount of instructions for nearly the same thing every time.
Since all projects (except npp) have installation instruction within their repo, i have shortened those sections to 2 or 3 sentences per editor.
In my view this is a better readable solution now.

And some idea to discuss:
I think the first explanation text is a bit duplicate to many informations that are told later anyways within the editor sections.

Maybe we could remove

> There are two bigger projects for this. 
_Eclipse Smart Home - Designer_ as a standalone editor and _openHAB VS Code_ as an extension for the Visual Studio Code open source editor.
Additionally some syntax highlighting solutions for other text editors are presented in the _Other Syntax Highlighting Solutions_ section.

completely.

Wdyt?

ThomDietrich commented:
Excellent 🍻 Looks good now. I like the additions to the VSCode project and what you did with "Others".
I wonder if we could go full gif and add one animated image to the designer as well. Would you be able to build one (I can recommend http://www.screentogif.com for Windows)?

One last request: You went with "Other Syntax Highlighting Solutions". The article is about Editors, not highlighting. An Editor Integration can bring more than just Highlighting and while all of them are only about Highlighting, I'd not reduce it to this.

Additionally to make the headline stable to be linked at could you define the anchor explicitly? See: https://github.com/openhab/openhab-docs/blame/gh-pages/installation/openhabian.md#L344

Last thing related to that: The link http://docs.openhab.org/installation/designer.html will not work any longer. that's bad. Could you add a redirect for that purpose? See and copy: https://github.com/openhab/openhab-docs/blob/gh-pages/features/sitemap.md

That's the last from me. 😃

Confectrician commented:
Hey @ThomDietrich,

Thanks for your response.
I will take care of those changes tommorow and try to get all done.

kubawolanin commented:
> (I can recommend http://www.screentogif.com for Windows)?

[LICECap](https://www.cockos.com/licecap/) masterrace 😎

Confectrician commented:
I use ShareX installed locally for such things. :)

ThomDietrich commented:
Let's fight ^^

Confectrician commented:
All changes done except the designer .gif
I will do that later, since i think we have to improve the section about the designer too then.
(It is currently written around those 3 screenshots.)

If the .gif is the only open thing for you guys now:
-> LGTM

Thanks for that constructive teamwork and the feedback within the last weeks. :)

ThomDietrich commented:
Hey @Confectrician, there was a slight misunderstanding regarding the anchors. I've committed that small change myself. Adding the gif later is fine with me. Let's merge ⚡️ As always thanks for the addition!

Confectrician commented:
Hey @ThomDietrich,

thx for merging this.

About your commit:
Yes i had misunderstood that due to the url which had a specific line mentioned.
I have changed 3 pages where the esh designer was mentioned and they will have a wrong anchor now, since this was before your anchor commit.

I will fix them in a separate pull request afterwards.

kubawolanin commented:
Hey guys, congrats on getting it merged already :)

I can't see the changes on the website though:
http://docs.openhab.org/configuration/index.html

Should it be visible already?

Confectrician commented:
Hey @kubawolanin,

I think the site simply hasnt been regenerated yet. :)
The files are definetely in the gh-pages branch and i think they will be included on the next serverside pagebuild.

ThomDietrich commented:
> have a wrong anchor now, since this was before your anchor commit.

@Confectrician I'm sorry about that. That's why we do reviews and I broke with that principle in the last second 🙈 

There was a problem with the last build. Please add the mentioned PR so we can trigger regeneration by merging it ;)

ThomDietrich commented on configuration/editors.md:
```diff
@@ -0,0 +1,122 @@
+Currently there are several existing solutions, that can help you configuring your openHAB instance in a textual way.
+This documentation page can give you some guidance in choosing the right one for you and setting it up.
+
+{::options toc_levels="2"/}
```
@Confectrician I didn't see that before. The default is `2..4` which would look better here, do you agree?

![grafik](https://user-images.githubusercontent.com/2870104/29773271-ec1e930e-8bfc-11e7-8de6-532e15fa53e4.png)

Confectrician replied:
Was a relict from the old "others" design. i have changed it and it comes with the link fix PR.
